plugins/database/redshift/redshift_test.go

```diff
@@ -0,0 +1,524 @@
+package redshift
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/hashicorp/vault/sdk/database/dbplugin"
+	"github.com/hashicorp/vault/sdk/helper/dbtxn"
+	"github.com/lib/pq"
+)
+
+/*
+To run these sets of acceptance tests, you must pre-configure a Redshift cluster
```

Comment:
Just curious, since Redshift is similar to Postgres, do you think these tests might pass against a Postgres Docker image? Doesn't have to be something we do in this PR, could be a task during the testing cycle before the beta.

Reply:
I would love it if they did, I spent a day trying to make that work. Unfortunately, PG 8 is substantially different than modern postgres versions. Making this even more difficult is that Redshift has it's own nuances in SQL that conflict with the postgres 8 variant it's built on, making even postgres 8 docker images unusable in mocking Redshift. 

I did try with a PG 8 image FWIW, this testing decision was the last resort option.

Reply:
Well, it was a beautiful thought!